### PR TITLE
Add a sample to capture if build was run on an Apple ARM processor (M1)

### PR DESCRIPTION
**Pull request by jprinet**

### CODE REVIEW

**Comment by jprinet** [comment]
when merging this, we'll have to thanks the initial author from the community and/or add him as co-author of this PR

**Review by runningcode — COMMENTED** [review]
this looks good, can we add a Maven sample as well?

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  def osArch = System.getProperty("os.arch")
+  api.value("os.arch", osArch)
+
+  if (isDarwin(osName)) {
+    if (isM1()) {
+      api.tag("M1")
+    }
+    if (isTranslatedByRosetta()) {
+      api.tag("proc_translated")
```
[the remark this answers is not in the document]
Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:32]: Not sure if we want to be so specific, the system variable being `sysctl.proc_translated`
what if the translator is named `Rosetta` in the future?

[the remark this answers is not in the document]
Comment (etiennestuder) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:32]: I have no expertise in this domain, but if we can use a more"user-friendly" name than proc_translated, that would be good, IMO.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  api.value("os.name", osName)
+
+  def osArch = System.getProperty("os.arch")
+  api.value("os.arch", osArch)
+
+  if (isDarwin(osName)) {
+    if (isM1()) {
+      api.tag("M1")
+    }
+    if (isTranslatedByRosetta()) {
```
Comment (etiennestuder) [inline build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:31]: Is this an `if` on purpose or should it be a `else if`?

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    def finished = process.waitFor(10, TimeUnit.SECONDS)
+    finished && process.exitValue() == 0 ? trimAtEnd(standardText) : null
+  } finally {
+    process.destroyForcibly()
+  }
+}
+
+static String trimAtEnd(String str) {
+  ('x' + str).trim().substring(1)
+}
```
Comment (etiennestuder) [inline build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:66]: Empty line at the end

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  return execAndGetStdout("uname", "-p") == "arm"
+}
+
+// On Apple silicon, a universal binary may run either natively or as a translated binary
+// https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment#Determine-Whether-Your-App-Is-Running-as-a-Translated-Binary
+static boolean isTranslatedByRosetta() {
+  return execAndGetStdout("sysctl", "sysctl.proc_translated") == "sysctl.proc_translated: 1"
+}
+
+static String execAndGetStdout(String... args) {
```
Comment (etiennestuder) [inline build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:51]: Is this implementation how we read from an Exec command in other places?

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 2 above)
Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:31]: It's an `if` on purpose as I'm unsure of the behavior of `uname -p` when Rosetta comes into play.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 4 above)
Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:51]: Yes, but those files are meant to be standalone given the header instructions:
`apply from: file('gradle-processor-arch.gradle')`

I'm ok factorizing but my opinion is that a sample should be self sufficient, wdyt?

Comment (runningcode) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:51]: I think Etienne's comment refers to the fact that this is not consistent with the CCUD plugin's way of doing this.
It is, however, consistent with the other sample scripts so I propose keeping this PR as is and doing a follow up PR to make the samples consistent with CCUD.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 2 above)
Comment (etiennestuder) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:31]: Please decide with @runningcode. I'd like to stay out of the details.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 4 above)
Comment (etiennestuder) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:51]: Yes, I was after consistency between scripts, not to share the code. All good it seems.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 2 above)
Comment (runningcode) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:31]: I think the `if else` would clarify the intention here if logically only one tag could be applied.

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (runningcode) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle]: 👍

File: build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle (hunk 2 above)
Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:28]: just done

Comment (runningcode) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:28]: Just double checking here Jerome: The link you sent describes the behavior of `uname -m` not `uname -p`. Can you ask someone on the team to test this with and without translation enabled?

Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle:28]: `uname -p` would just hides a few details but lead to the same conclusion
https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line

Script has already been tested (not translated yet)

**Pull request merged**